### PR TITLE
feat: add libcbor

### PR DESCRIPTION
**Pull request by gocarlos**
Specify library name and version:  **libcbor/0.6.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by gocarlos** [comment]
clang builds probably failing due to https://github.com/PJK/libcbor/issues/140, latest tag still does not have this fix

**Comment by gocarlos** [comment]
> clang builds probably failing due to [PJK/libcbor#140](https://github.com/PJK/libcbor/issues/140), latest tag still does not have this fix

@PJK could you create a new tag e.g. 0.6.2 which the fix mentioned above?

File: recipes/libcbor/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+    generators = "cmake", "cmake_find_package"
```
Comment (jgsogo) [inline recipes/libcbor/all/conanfile.py:23]:
```suggestion
    generators = "cmake"
```

There aren't requirements, so nothing to consume using `find_package`

**Comment by SSE4** [comment]
@danimtb seems like CI stucks

**Pull request merged**